ratatool-scalacheck/README.md

```diff
@@ -39,7 +37,50 @@ Implicit Arbitrary instances are also available for Avro and Protobuf records. E
 
 ```scala
 val avroArb: Arbitrary[MyRecord] = implicitly[Arbitrary[MyRecord]]
+```
+
+Multiple `Gen` instances can be composed together with a for-comprehension. This will cause the data to be generated with the same seed.
+```scala
+val genData: Gen[(MyRecord, Int)] = for {
+ record <- avroGen
+ i <- Gen.choose(0, 10)
+} yield (record, i)
+
+// .sample gets a random seed
+val data: Option[(MyRecord, Int)] = genData.sample
+```
+
+Due to scalacheck internals generation can fail so the return type of `.sample` is `Option[A]`.
+In general, calling `.get` on an `Option` is an anti-pattern because this can produce `NoSuchElementException`.
+When used in conjunction with scalacheck, calling `.sample.get` can be the root cause of flaky tests.
+
+Repeatedly calling `.sample` is also an anti-pattern as each generated instance will be produced with a different random seed and tests will not be reproducible.
+Instead, use `withGen` as described below.
 
+## In tests
+
+When using `Gen` to create data in a test, use `withGen` to capture the random seed and print it on failure:
```

Comment:
Again I think we can reword this to be less suggestive. We don't want to tell users _what_ to do, just _how_ to do different things and enable them to make their own decisions